yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1839,6 +1828,34 @@ def off_axis_projection_SPH(np.float64_t[:] px,
         px_rotated[i] = rotated_coordinates[0]
         py_rotated[i] = rotated_coordinates[1]
 
+    return px_rotated, py_rotated, rot_bounds_x0, rot_bounds_x1, rot_bounds_y0, rot_bounds_y1
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def off_axis_projection_SPH(np.float64_t[:] px,
+                            np.float64_t[:] py,
+                            np.float64_t[:] pz,
+                            np.float64_t[:] particle_masses,
+                            np.float64_t[:] particle_densities,
+                            np.float64_t[:] smoothing_lengths,
+                            bounds,
+                            center,
+                            width,
+                            np.float64_t[:] quantity_to_smooth,
+                            np.float64_t[:, :] projection_array,
+                            normal_vector,
+                            north_vector,
+                            weight_field=None):
+    # Do nothing in event of a 0 normal vector
+    if np.allclose(normal_vector, np.array([0., 0., 0.]), rtol=1e-09):
```

Comment:
I guess this could be simplified ? (relative tolerance has no effect when comparing to 0)
```suggestion
    if np.allclose(normal_vector, 0.):
```

Reply:
W_iiii_

Reply:
I refactored a function and this bit of code came over from that, so we've had it for a while. I can change it, though.